src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -677,8 +699,155 @@ namespace OpenMS
       return EXECUTION_OK;
     }
 
-protected:
-    struct PeptideProteinMatchInformation
+ protected:
+
+     using MapDecoyStringToPrefixSuffixCount = std::map<std::string, std::pair<int, int>>;
+     using MapCaseInsensitiveToCaseSensitiveDecoyString = std::map<std::string, std::string>;
+
+     template<typename T>
+     ExitCodes findDecoyString_(FASTAContainer<T> & proteins,
```

Comment:
These are quite useful functions, and I think they deserve their own class. I would not associate them with PeptideIndexer.
Since `determineDecoyStringFromCounts` is called only once, you could think about merging the functionality into `findDecoyString()` and save some boilerplate code since many things can be counted/computed when iterating over the proteins already.

Reply:
Fine with me. 
If any other people think that this deserves its own class, then I'll gladly outsource it.